src/test/munit/sftp-write-test-case.xml

```diff
@@ -114,4 +114,53 @@
         </munit:execution>
     </munit:test>
 
+    <flow name="write-number-collection-until-successful">
+        <choice>
+            <when expression="#[payload == 0]">
+                <set-variable variableName="writeMode" value="CREATE_NEW" />
+            </when>
+            <otherwise>
+                <set-variable variableName="writeMode" value="APPEND" />
+            </otherwise>
+        </choice>
+        <set-payload value="#[payload as String]"/>
+        <until-successful maxRetries="10" millisBetweenRetries="1000">
+            <sftp:write config-ref="${config}" path="filePath.txt" lock="true" mode="#[vars.writeMode]">
+                <sftp:content>#[payload]</sftp:content>
+            </sftp:write>
+        </until-successful>
+    </flow>
+
+    <munit:test name="consecutive-writes-from-separate-flows-succeed-with-until-successful">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="write-number-collection-until-successful"/>
+        </munit:enable-flow-sources>
+        <munit:behavior>
+            <set-variable variableName="fileName" value="filePath.txt"/>
+            <set-variable variableName="fileSize" value="#[6]"/>
+        </munit:behavior>
+        <munit:execution>
+            <foreach collection="#[1 to vars.fileSize]">
```

Comment:
The var fileSize is used to count iterations and then for fileSize assertion. Should be two separated vars.